journal.py

```diff
@@ -27,6 +31,20 @@
 )
 
 
+def style_text(text):
+    output = ""
+    if text is None:
+        pass
+    else:
+        text = text.split("```")
+        for i in range(len(text)):
+            if i % 2 == 0:
+                output += markdown.markdown(text[i])
+            else:
+                output += highlight(text[i], PythonLexer(), HtmlFormatter())
+    return output
```

Comment:
You are working much to hard for this.  Try using the markdown library extensions:

```python
markdown(text, extensions=['codehilite', 'fenced-code'])
```